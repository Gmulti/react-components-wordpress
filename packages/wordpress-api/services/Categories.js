import Endpoint from './shared/Endpoint'

class Categories extends Endpoint {
    get endpoint() {
        return '/categories'
    }
}

export default Categories
