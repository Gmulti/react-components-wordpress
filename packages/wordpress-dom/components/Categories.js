import React from 'react'

function Categories({ items = [] }) {
    return (
        <ul>
            {items.map((item, key) => (
                <li key={key}>{item.name}</li>
            ))}
        </ul>
    )
}

export default Categories
