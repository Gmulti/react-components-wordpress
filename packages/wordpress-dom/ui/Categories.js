import React from 'react'
import PropTypes from 'prop-types'

function Categories({ items }) {
    return (
        <ul>
            {items.map((item, key) => (
                <li key={key}>{item.name}</li>
            ))}
        </ul>
    )
}

Categories.propTypes = {
    items: PropTypes.array
}

export default Categories
