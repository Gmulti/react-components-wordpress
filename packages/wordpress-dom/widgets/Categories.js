import React from 'react'
import { connectCategories } from '../../wordpress-api/connectors/connectCategories'
import CategoriesComponent from '../components/Categories'

const Categories = props => <CategoriesComponent {...props} />

export default connectCategories(Categories)
