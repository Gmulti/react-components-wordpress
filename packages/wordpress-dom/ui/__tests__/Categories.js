import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'
import Categories from '../Categories'

Enzyme.configure({ adapter: new Adapter() })

describe('Categories', () => {
    it('expect to render', () => {
        const wrapper = shallow(<Categories />)

        expect(wrapper).toMatchSnapshot()
    })

    it('expect to render with props', () => {
        const instance = renderer.create(
            <Categories
                items={[
                    {
                        name: 'Category 1'
                    },
                    {
                        name: 'Category 2'
                    }
                ]}
            />
        )

        expect(instance.toJSON()).toMatchSnapshot()
    })
})
