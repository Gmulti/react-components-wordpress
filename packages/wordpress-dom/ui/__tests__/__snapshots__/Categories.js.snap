// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories expect to render 1`] = `ShallowWrapper {}`;

exports[`Categories expect to render with props 1`] = `
<ul>
  <li>
    Category 1
  </li>
  <li>
    Category 2
  </li>
</ul>
`;
